HedvigInsurance/embark
# fix: whitelist files that should be included into embark package

## What

Whitelist that should be included into `embark` package by using `package.json` `files` field;

## Why

We are currently including all the files which might end up being a security issue:
<img width="1604" alt="all-files" src="https://user-images.githubusercontent.com/19200662/152997424-b0488391-8049-42e3-a09f-b78b868607a6.png">

There's a `.npmignore` file at the root of `embark` package that were suppose to avoid this issue as it's declaring that everything rather then `dist/` should be ignore while composing the package. But it seems that, [when placed in the root of the project](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#:~:text=At%20the%20root%20of%20your%20package%20it%20will%20not%20override%20the%20%22files%22%20field%2C%20but%20in%20subdirectories%20it%20will.), that file will not override `package.json` `files` field [which is `[*]` by default (include everything)](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#:~:text=Omitting%20the%20field%20will%20make%20it%20default%20to%20%5B%22*%22%5D%2C%20which%20means%20it%20will%20include%20all%20files.).

One can use `npm pack` or `npx npm-packlist` in order to check the files that will be included in the `embark` package. Below you can see the results of `npm pack` command before and after this new setup.

**Before**
<img width="844" alt="before" src="https://user-images.githubusercontent.com/19200662/152999471-00baf1ce-80c8-4b29-a6ca-32303b6241c7.png">

**After**
<img width="844" alt="after" src="https://user-images.githubusercontent.com/19200662/152999500-d6f96075-4010-4733-91e3-6b33caa53dbc.png">

## How to test

Here's how I've tested:

1. While on this feature branch, generate a new package by running `npm pack` (you can take this opportunity to also double check the files that were included into the package since they are logged during the execution of this command);
2. Go to `web-onboarding` and reinstall `embark` using the package generated in the previous step: `yarn add <path-to-package.tgz>`. At the end of this command you can check that `embark` dependencies under `node_modules/@hedviginsurance` has only the files listed on step 1;
3. Run `web-onboarding` locally with `yarn watch`. You should be able to complete _Embark_ flows as before;
